sdk/spring/spring-messaging-azure-eventhubs/src/test/java/com/azure/spring/messaging/eventhubs/core/EventHubsTemplateTests.java

```diff
@@ -78,6 +80,35 @@ void testSendAsyncForMessagesWithThreeBatch() {
         verify(this.mockProducerClient, times(3)).send(any(EventDataBatch.class));
     }
 
+    /**
+     * test the three batches case in parallel
+     */
+    @Test
+    void testSendAsyncForMessagesWithThreeBatchParallel() {
+        EventDataBatch eventDataBatch = mock(EventDataBatch.class);
+
+        when(this.mockProducerClient.createBatch(any(CreateBatchOptions.class)))
+            .thenReturn(Mono.just(eventDataBatch));
+        when(eventDataBatch.tryAdd(any(EventData.class))).thenReturn(true, true, false, true, true,
+            false, true);
+        when(eventDataBatch.getCount()).thenReturn(2, 2, 1);
+        List<String> messagesList = new ArrayList<>();
+        for (int i = 0; i < 5; i++) {
+            messagesList.add("abcde");
+        }
+        List<Message<String>> messages =
+            messagesList.stream().map((Function<String, GenericMessage<String>>) GenericMessage::new).collect(Collectors.toList());
+
+        Mono<Void> mono = Flux.just(messages)
```

Comment:
Should we use StepVerifier for reactor testing?

Reply:
Yes, used.